redash/models.py

```diff
@@ -500,8 +500,10 @@ def to_dict(self, all=False, with_permissions=False):
             d['scheduled_queue_name'] = self.scheduled_queue_name
             d['groups'] = self.groups
 
-        if with_permissions:
-            d['view_only'] = self.data_source_groups.view_only
+        if with_permissions_for is not None:
+            d['view_only'] = DataSourceGroup.get(
```

Comment:
This way we need to make another database query, while we already have the object from calling `add_group`/`remove_group`. How about we change those methods to return the `DataSourceGroup` object, and then assign the `view_only` value in the handler code?